subprojects/plugins/src/test/groovy/org/gradle/api/tasks/testing/TestTaskSpec.groovy

```diff
@@ -264,43 +264,52 @@ class TestTaskSpec extends AbstractProjectBuilderSpec {
         0 * closure._
     }
 
-    def "adds listeners and removes after execution"() {
+    def "removes listeners after execution"() {
```

Comment:
💭 Do you need to test `AbstractTestTask#addTestOutputListener(TestOutputListener)` in the same way?

Reply:
Yeah, we could and maybe should.

Reply:
I couldn't figure how to do that. Couldn't get the event to be observed. Any tips?